packages/account/src/providers/transaction-response/transaction-response.ts

```diff
@@ -78,10 +83,40 @@ export type TransactionResultReceipt =
 
 /** @hidden */
 export type TransactionResult<TTransactionType = void> = TransactionSummary<TTransactionType> & {
-  gqlTransaction: GqlTransaction;
   logs?: Array<unknown>;
 };
 
+function mapGqlOutputsToTxOutputs(outputs: GqlMalleableFieldsFragment['outputs']): Output[] {
+  return outputs.map((o) => {
+    const obj = 'amount' in o ? { ...o, amount: bn(o.amount) } : o;
+    switch (obj.type) {
+      case 'CoinOutput':
+        return { ...obj, type: OutputType.Coin } satisfies OutputCoin;
+      case 'ContractOutput':
+        return {
+          ...obj,
+          type: OutputType.Contract,
+          inputIndex: parseInt(obj.inputIndex, 10),
+        } satisfies OutputContract;
+      case 'ChangeOutput':
+        return {
+          ...obj,
+          type: OutputType.Change,
+        } satisfies OutputChange;
+      case 'VariableOutput':
+        return { ...obj, type: OutputType.Variable } satisfies OutputVariable;
+      case 'ContractCreated':
+        return {
+          ...obj,
+          type: OutputType.ContractCreated,
+          contractId: obj.contract,
+        } satisfies OutputContractCreated;
+      default:
+        return assertUnreachable(obj);
```

Comment:
The `assertUnreachable` will throw a compile-time error if the switch doesn't exhaust all types in the union, and thus won't allow a build at all. I think it's a _very_ useful thing to have for all switches. I could put the error you mentioned in `assertUnreachable`, though I'm not sure if we should do it because, in my mind, `FuelError` should be related to domain-specific errors, not stuff like unreachable code.

Reply:
This is magic - I approve!